lms/resources/lti_launch.py

```diff
@@ -24,21 +23,17 @@ def __init__(self, request):
         """Return the context resource for an LTI launch request."""
         self._request = request
         self._authority = self._request.registry.settings["h_authority"]
-        self._application_instance_service = self._request.find_service(
-            name="application_instance"
-        )
         self._assignment_service = request.find_service(name="assignment")
 
     def get_or_create_course(self):
         """Get the course this LTI launch based on the request's params."""
         course_service = self._request.find_service(name="course")
         params = self._request.parsed_params
 
-        tool_consumer_instance_guid = params["tool_consumer_instance_guid"]
         context_id = params["context_id"]
 
         authority_provided_id = course_service.generate_authority_provided_id(
-            tool_consumer_instance_guid, context_id
+            self.js_config.application_instance.tool_consumer_instance_guid, context_id
```

Comment:
This change is in a separate commit.

Instead of using the parameter from the request use it directly form the application instance which is now guaranteed to have the same one:


https://github.com/hypothesis/lms/blob/5140598c77655437695d09166fc7d7947043434e/lms/models/application_instance.py#L175


This simplifiies things a bit for LTI1.3

Reply:
I know that the `js_config` happens to have an application instance, but what has the application instance got to do with `js_config`? Are we just using this because it happens to have one?

Seems a bit weird.